source/blog/2015-05-05-glimmer-merging.md

```diff
@@ -0,0 +1,155 @@
+  continued performance improvements in Ember throughout the 2.x cycle
+  as a result of this change.
+
+## Performance Gains
+
+The biggest performance gains in Glimmer come from moving to a simpler
+rendering model, built on top of HTMLBars.
+
+First, this allowed us to remove **all** internal views for constructs
+like `{{foo}}`, `{{#if bar}}` and even in `{{#each posts as |post|}}`.
+This view removal has an impact on initial render, because these
+constructs are very common in real-world template.
+
+Second, as we have discussed extensively, this allows us to
+significantly improve the performance of re-rendering, which makes it
+practical to re-render lists with entirely new arrays with very good
+performance. Previously, achieving reasonable performance was very
+difficult, and even when possible, came with significant bookkeeping
+overhead.
+
+Interestingly, we have found that when testing real apps, the
+performance gains are much more widespread than we expected, in large
+part because of the simplification of the overall model.
+
+**Glimmer's performance in real applications has pleasantly surprised
+us, exceeding the improvements we've seen in benchmarks designed to
+stress-test pathological cases.**
+
+When upgrading to Glimmer, please pay special attention to the
+real-world performance of your application **in production mode** and
+after clearing any deprecations with performance warnings.
+
+## Deprecated Features
+
+Throughout the 1.x series, Ember has deprecated features we intended to
+remove in 2.0. That process has continued with Ember 1.13, which will
+contain Glimmer.
+
+However, it's worth noting that Glimmer is the first major change to
+many parts of the view since Ember was called SproutCore 2! As a result,
+perfect compatibility, especially in private features, was more
+challenging.
+
+During the process of building Glimmer, we found various semantics of
+"controller"s to be the most challenging. For the most part, this is
+because the concept of controller has a number of different meanings
+depending on context (routes, `{{render}}`, `{{#each posts
+itemController='post'}}`, `{{#each posts itemController='post' as
+|post|}}`, `{{#with someController}}`, etc.).
+
+Controllers both controller the template's "context" (called "self" in
+Glimmer) and serve as a target for actions, but the specific details
+vary, and the semantics are effectively derived from implementation
+details of Ember 1.x's rendering engine.
+
+Glimmer, through HTMLBars, has a much clearer concept of "scope", and
+features like `{{yield}}` work directly with that scope object. We were
+able to get the Ember test suite passing by implementing the old
+semantics on top of the new scope concept, but **we are aware that we
+have gaps in our implementation.**
+
+Please let us know if you find controller semantics that we have
+implemented incorrectly. Bug reports would help, JSBins would help more,
```

Comment:
Yes, I have been using http://emberjs.jsbin.com/rwjblue/413/edit?html,js,output for a starter while testing.  Once we merge, the canary build will be the correct path so http://emberjs.jsbin.com/rwjblue/397/edit is a better starting point.